R/methods-epi_df.R

```diff
@@ -80,13 +80,39 @@ summary.epi_df = function(object, ...) {
   } 
   
   cn <- names(res)
-  nr <- vctrs::vec_size(x) 
-  not_epi_df <- (!("time_value" %in% cn) || !("geo_value" %in% cn) || vctrs::vec_size(res) > nr || any(i > nr))
   
-  if (not_epi_df) return(tibble::as_tibble(res))
+  # Duplicate columns, Abort
+  dup_col_names = cn[duplicated(cn)]
+  if (length(dup_col_names) != 0) {
+    Abort(paste0("Column name(s) ", 
+                 paste(unique(dup_col_names), 
+                       collapse = ", "), " must not be duplicated."))
+  } 
+  
+  # If not an epi_df (missing time_value or geo_value), return a tibble
+  not_epi_df <- (!("time_value" %in% cn) || !("geo_value" %in% cn))
   
-  # Use reclass as safeguard (in case class & metadata are dropped)
+  if (not_epi_df) {
+    attributes(res)$metadata <- NULL 
+    return(tibble::as_tibble(res))
+  }
+  
+  # Use reclass as safeguard (in case class &/or metadata are dropped)
   reclass(res, attr(x, "metadata"))
```

Comment:
I don't think this will actually alter `res`; you probably need `res <- reclass(res, attr(x, "metadata"))`